### PR TITLE
challenge/parking lot

### PR DESCRIPTION
**Pull request by schutza**
Java coding challenge to rekindle myself with modern Java constructs.

### CODE REVIEW

File: src/main/java/ie/textability/kata/parking/ParkingLot.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+    public ParkingLot withMotorcycle() {
+        System.out.println("Accommodating "+ VehicleType.MOTORCYCLE +": " + occupiedSpace.get(ParkingSpotType.COMPACT).size());
+        if (occupiedSpace.get(ParkingSpotType.COMPACT).size() < maxCapacity.get(ParkingSpotType.COMPACT)) {
+            occupiedSpace.get(ParkingSpotType.COMPACT).add(VehicleType.MOTORCYCLE);
+        } else if (occupiedSpace.get(ParkingSpotType.REGULAR).size() < maxCapacity.get(ParkingSpotType.REGULAR)) {
+            occupiedSpace.get(ParkingSpotType.REGULAR).add(VehicleType.MOTORCYCLE);
+        } else if(occupiedSpace.get(ParkingSpotType.LARGE).size() < maxCapacity.get(ParkingSpotType.LARGE)) {
+            occupiedSpace.get(ParkingSpotType.LARGE).add(VehicleType.MOTORCYCLE);
+        } else {
+            throw new ParkingLotFullException("FULL");
```
Comment (schutza) [inline src/main/java/ie/textability/kata/parking/ParkingLot.java:62]: need unit tests for `withXYZ` methods - especially around exceptions and overflow logic

File: src/main/java/ie/textability/kata/parking/ParkingLot.java [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        return occupiedSpace.get(ParkingSpotType.REGULAR);
+    }
+
+    public List<VehicleType> getLargeSpots() {
+        return occupiedSpace.get(ParkingSpotType.LARGE);
+    }
+
+    /*
+     * Business logic
+     */
```
Comment (schutza) [inline src/main/java/ie/textability/kata/parking/ParkingLot.java:135]: business logic is just a pass-through for CapacityAnalyser. It is probably not needed here, and would reduce coupling.

File: src/main/java/ie/textability/kata/parking/ParkingSpotType.java [hunk 3]
```diff
@@ -0,0 +1,5 @@
+package ie.textability.kata.parking;
+
+public enum ParkingSpotType {
```
Comment (schutza) [inline src/main/java/ie/textability/kata/parking/ParkingSpotType.java:3]: might embed the enums into ParkingLot

File: src/main/java/ie/textability/kata/parking/VehicleType.java [hunk 4]
```diff
@@ -0,0 +1,7 @@
+package ie.textability.kata.parking;
+
+public enum VehicleType {
```
Comment (schutza) [inline src/main/java/ie/textability/kata/parking/VehicleType.java:3]: might embed this enum into ParkingLot

**Review by schutza — COMMENTED** [review]
add unit tests for ParkingLot

File: src/main/java/ie/textability/kata/parking/ParkingSpotType.java (hunk 3 above)
Comment (schutza) [reply src/main/java/ie/textability/kata/parking/ParkingSpotType.java:3]: leaving as first-class citizen
